### PR TITLE
184 fix codeblock styles

### PR DESCRIPTION
**Pull request by ericvalcik**
Addressing issue #184.

Please check out the code styling - hopefully everything looks as similarly as possible to the codeblock outside of mdx.

I also tried to tackle the duplication of content inside the `component/[slug].mdx` files, but it turned out to be trickier than I thought.

Most difficult was the interpolation of variables inside the triple-quoted code segment in the `mdx` files - I found a _hack_ to get it working in [nextra](https://github.com/shuding/nextra/blob/bf2636a20a7b81a850476a549b73e1b6dc20c278/docs/pages/docs/guide/syntax-highlighting.mdx#L229-L281). I also needed to create a new file for the `DynamicCode` component as that had to be a client component and because I extracted it from a `mdx` file I chose to save it as a `JSX` component not a `TSX` (the only place we use it is inside a mdx).

### CODE REVIEW

**Review by iamhectorsosa — COMMENTED** [review]
Not sure if the solution is working. I see the variable name being rendered in the component when looking at the preview deployment.

**Review by iamhectorsosa — COMMENTED** [review]
Almost there! This looks a bit more consistent:
image


This CSS seems missing. Apply as CSS or as Tailwind utility classes as you see fit.

```
element.style {
    font-size: .875rem;
    line-height: 1.5rem;
    padding-block: 1.5rem; // py-6
}
```

...for font size and height careful with the text utility as they combine both of these.

**Comment by ericvalcik** [comment]
Good catch, the `padding-y` is actually identical - only handled a little differently as the numbering has to be done in a different way. Even the `word-wrap`-ing should be identical, but again, had to be done in a slightly different way than the TSX version.

**Comment by ChcJohnie** [comment]
Nice work, from eye test everything looks fine, just copy button is slightly mismatched (noticeable when you switch between auto and manual, but no idea whats cause from devtool)
OT suggestion, but with proper styling I noticed that fade for collapsed codeblock could be more blueish (very subjective though)

Also I am not sure if I am comfortable with full component page mdx shared as I still feel there might be more of component specific content (thing I can think of right now is component API reference). Should be pretty okay to reuse installation. But maybe there is no more component specific content and full reuse is then fine. wdyt?

**Comment by ChcJohnie** [comment]
Btw another bug/bad-feature note , I see Mdx code tab overflow is fixed. 
But code copy function doesnt have good UX together with CLI content of both command and options doc in one tab. Imho simple fix would be to divide into two block (one code, one non-code?) but not sure about its visual

**Comment by ericvalcik** [comment]
@iamhectorsosa you are right, I misread `padding-y` for `padding-x` -> fixed that

@ChcJohnie copy button is fixed (at least alignment). I agree that currently the copy button in the following screenshot doesn't make sense, best would be change the structure or even remove the copybutton completely in this particular case -> I think that maybe that could be done with mdx, but not sure.

image

**Comment by iamhectorsosa** [comment]
I don't want to remove the copy button anywhere. In any case we can grab options and use a proper bullet list below with markdown.

**Review by iamhectorsosa — DISMISSED** [review]

File: src/docs/content/utils/DynamicCode.jsx [hunk 1]
```diff
@@ -0,0 +1,18 @@
+"use client";
```
Comment (iamhectorsosa) [inline src/docs/content/utils/DynamicCode.jsx:1]:
```tsx
import * as React from "react";

export function DynamicCode({ textContent, installName, children }: {textContent: string, installName: string, children: React.ReactNode}) {
  const ref = React.useRef<HTMLDivElement | null>(null);

  React.useEffect(() => {
    if (ref.current) {
      const elements = ref.current.querySelectorAll("span.line span");
      elements.forEach((el)=> {if(el.textContent === textContent){
        el.textContent = ` [options] ${installName}`;
      }})
    }
  }, [installName, textContent]);
  
  return <div ref={ref}>{children}</div>;
}
```

Comment (iamhectorsosa) [reply src/docs/content/utils/DynamicCode.jsx:1]: This is very very subject to how Shiki parses that code

File: src/docs/components/MdxComponents/DocElements.tsx [hunk 2]
```diff
@@ -103,3 +103,22 @@ export const MdxUl = ({ children }: React.ComponentPropsWithoutRef<"ul">) => (
     {children}
   </ul>
 );
+export const MdxH4 = (props: React.ComponentPropsWithoutRef<"h4">) => {
+  const { children, id, className, ...otherProps } = props;
+  return (
+    <h3
```
Comment (ChcJohnie) [inline src/docs/components/MdxComponents/DocElements.tsx:109]: Is h3 intentional here?

File: src/docs/content/utils/shared.mdx [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+      <TabsList className="mt-8 rounded-2xl">
+        <TabsTrigger value="automatic" className="px-7 py-2 rounded-[10px]">Automatic Install</TabsTrigger>
+        <TabsTrigger value="manual" className="px-7 py-2 rounded-[10px]">Manual Install</TabsTrigger>
+      </TabsList>
+    </div>
+
+    <TabsContent value="automatic">
+      <DynamicCode replaceBy={props.installName}>
+        ```shell
+        npx @mailingui/cli add [options] installName
```
Comment (ChcJohnie) [inline src/docs/content/utils/shared.mdx:38]: Imho much better, I would probably also remove [options] in sake of runnable copy code, but clarity approach also fine to me.

**Review by ChcJohnie — APPROVED** [review]

File: src/docs/content/utils/shared.mdx (hunk 3 above)
Comment (ericvalcik) [reply src/docs/content/utils/shared.mdx:38]: I read this comment after I pushed the changes, so I left it as is for now. Let me know if you'd like to change it

File: src/docs/content/utils/DynamicCode.jsx (hunk 1 above)
Comment (ericvalcik) [reply src/docs/content/utils/DynamicCode.jsx]: yes, it indeed is, but if we only use it here I feel like it is worth it than just having the same text on 8 places.

The code is updated now

**Comment by iamhectorsosa** [comment]
Closing as irrelevant

**Pull request closed without merging**